table/block_based_table_reader.cc

```diff
@@ -1771,25 +1807,28 @@ TBlockIter* BlockBasedTable::NewDataBlockIterator(
   return iter;
 }
 
-Status BlockBasedTable::MaybeLoadDataBlockToCache(
+Status BlockBasedTable::ReadBlockAndMaybeLoadToCache(
```

Comment:
ReadBlockAndMaybeLoadToCache is misleading since we would not read the data if fill_cache is false. I think the old MaybeLoadDataBlockToCache is the most descriptive.

Reply:
How about MaybeReadBlockAndLoadToCache?

Reply:
That is better.